sources/Notifications/Repositories/YoutubeRepository.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading.Tasks;
+using Cdn.Repositories;
+using Database;
+using Database.Models.Entities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Notifications.Errors;
+using Remora.Results;
+
+namespace Notifications.Repositories
+{
+	public class YoutubeRepository : IYoutubeRepository
+	{
+		private readonly ArkadiaDbContext _context = new ArkadiaDbContext();
+
+		[ExcludeFromCodeCoverage]
+		public ValueTask DisposeAsync()
+		{
+			return _context.DisposeAsync();
+		}
+
+		public ValueTask<YoutubeSubscription?> GetSubscriptionByIdOrDefaultAsync(string id)
+		{
+			return _context.YoutubeSubscriptions.FindAsync(id);
+		}
+
+		public ValueTask<Guild?> GetGuildByIdOrDefaultAsync(string id)
+		{
+			return _context.Guilds.FindAsync(id);
+		}
+
+		public async Task ModifyExpiryAsync(string id, DateTime newTime)
+		{
+			var subscription = await GetSubscriptionByIdOrDefaultAsync(id);
+			
+			// leave this nullability warning, it is expected for the caller to check if it exists first
+			// if not, we'd like to throw anyway.
+			subscription!.ExpiresAt = newTime;
+
+			await _context.SaveChangesAsync();
+
+		}
+
+		public IEnumerable<YoutubeSubscription> GetSubscriptions()
+		{
+			return _context.YoutubeSubscriptions;
+		}
```

Comment:
This will force multiple synchronous `SELECT`'s (one per entry in the entire table) to happen.